flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/DynamicFilteringData.java

```diff
@@ -141,6 +144,24 @@ private int hash(RowData row) {
         return Objects.hash(Arrays.stream(fieldGetters).map(g -> g.getFieldOrNull(row)).toArray());
     }
 
+    public boolean equals(DynamicFilteringData another) {
```

Comment:
Might change it to a static method outside of the DynamicFilteringData if we do not want to make it to be Object#equals